app/scripts/lib/multichain-api/wallet-requestPermissions.test.js

```diff
@@ -151,283 +175,454 @@ describe('requestPermissionsHandler', () => {
+            value: ['0x1'],
+          },
+        ],
+      },
+    });
+  });
+
+  it('requests approval for eth_accounts and permittedChains when only permittedChains is specified in params', async () => {
+    const {
+      handler,
+      getNetworkConfigurationByNetworkClientId,
+      requestPermissionApprovalForOrigin,
+    } = createMockedHandler();
+
+    await handler({
+      ...getBaseRequest(),
+      params: [
+        {
+          [PermissionNames.permittedChains]: {
+            caveats: [
+              {
+                type: CaveatTypes.restrictNetworkSwitching,
+                value: ['0x64'],
+              },
+            ],
+          },
+        },
+      ],
+    });
+
+    expect(getNetworkConfigurationByNetworkClientId).not.toHaveBeenCalled();
+    expect(requestPermissionApprovalForOrigin).toHaveBeenCalledWith({
+      [RestrictedMethods.eth_accounts]: {},
+      [PermissionNames.permittedChains]: {
+        caveats: [
+          {
+            type: CaveatTypes.restrictNetworkSwitching,
+            value: ['0x64'],
+          },
+        ],
+      },
+    });
+  });
+
+  it('requests approval for eth_accounts and permittedChains when both are specified in params', async () => {
+    const {
+      handler,
+      getNetworkConfigurationByNetworkClientId,
+      requestPermissionApprovalForOrigin,
+    } = createMockedHandler();
+
+    await handler({
+      ...getBaseRequest(),
+      params: [
+        {
+          [RestrictedMethods.eth_accounts]: {
+            foo: 'bar',
+          },
+          [PermissionNames.permittedChains]: {
+            caveats: [
+              {
+                type: CaveatTypes.restrictNetworkSwitching,
+                value: ['0x64'],
+              },
+            ],
+          },
+        },
+      ],
+    });
+
+    expect(getNetworkConfigurationByNetworkClientId).not.toHaveBeenCalled();
+    expect(requestPermissionApprovalForOrigin).toHaveBeenCalledWith({
+      [RestrictedMethods.eth_accounts]: {
+        foo: 'bar',
+      },
+      [PermissionNames.permittedChains]: {
+        caveats: [
+          {
+            type: CaveatTypes.restrictNetworkSwitching,
+            value: ['0x64'],
+          },
+        ],
+      },
+    });
+  });
+
+  it('requests other permissions from params, but ignores CAIP-25 if specified', async () => {
     const { handler, requestPermissionsForOrigin } = createMockedHandler();
 
-    await handler(baseRequest);
+    await handler({
+      ...getBaseRequest(),
+      params: [
+        {
+          [Caip25EndowmentPermissionName]: {},
+          otherPermission: {},
+        },
+      ],
+    });
     expect(requestPermissionsForOrigin).toHaveBeenCalledWith({
-      eth_accounts: {},
       otherPermission: {},
     });
   });
 
-  describe('BARAD_DUR flag is not set', () => {
-    beforeAll(() => {
-      delete process.env.BARAD_DUR;
+  it('requests other permissions from params, but ignores eth_accounts if specified', async () => {
```

Comment:
wait what

Reply:
clearly missing something here...

Reply:
Is this just saying that the `requestPermissionsForOrigin` hook won't be called if permissions are not either eth_accounts or permittedChains? If so I feel like this is confusingly worded.

Reply:
Yeah I think I see now in the previous tests you specified `requests approval` and here you're saying `requests permissions`. I think this distinction may be a bit too subtle and may need a bit more explication to avoid confusion for others in the future

Reply:
the distinction to make is requesting approval vs requesting permissions, although this scenario needs to be reworded still since it is already causing confusion

Reply:
Yeah I think rewording would be helpful, since in practice requestingPermissions always also entails an approval and so its a subtle distinction that would require a lot of familiarity to grok